### PR TITLE
feat: AI Settings — live API-key verifier on the connect form

### PR DESCRIPTION
**Pull request by cyberk1ng**
## Summary by CodeRabbit

* **New Features**
  * Live, debounced API-key / Base URL verification on the AI provider connect form with checking spinner, verified state (model count + latency) and failed state with retry/re-test.

* **Localization**
  * Added localized strings for the verifier UI (status, success, failure details, retry/retest) across supported languages.

* **Tests**
  * Added comprehensive unit and widget tests for verifier UI, probe behaviors, timeouts and edge cases.

* **Documentation**
  * Changelog updated describing the new verifier.

* **Chores**
  * App version bumped for release.

[Review Change Stack](https://app.coderabbit.ai/change-stack/matthiasn/lotti/pull/3160)

### CODE REVIEW

File: lib/features/ai/ui/settings/services/connection_verifier_service.dart [hunk 1]
```diff
@@ -0,0 +1,362 @@
...
+      );
+      // Drop the result if a newer probe started while we were awaiting,
+      // or if the provider (auto-dispose) has been disposed in the meantime
+      // — e.g. the user navigated away from the connect form while a probe
+      // was in flight. Touching `state` on a disposed Ref throws.
+      if (myGen != _generation || !ref.mounted) return;
+      state = result;
+    } on Exception catch (e) {
+      if (myGen != _generation || !ref.mounted) return;
+      state = ConnectionCheckFailedNetwork(message: e.toString());
```
[the remark this answers is not in the document]
Comment (cyberk1ng) [reply lib/features/ai/ui/settings/services/connection_verifier_service.dart:261]: Fixed both halves of this. (1) **Default base URL** is now resolved up front: when `baseUrl.trim()` is empty, the verifier looks up `ProviderConfig.getDefaultBaseUrl(providerType)` (the same map the save path uses), so users on the official endpoint don't have to retype it to run the probe. (2) **ArgumentError** is now prevented at the source rather than caught — I added a stricter pre-probe gate that rejects URIs whose scheme isn't `http`/`https` or whose host is empty, returning `ConnectionCheckFailedNetwork` with an actionable message. `http.Client` only throws `ArgumentError` ("Unsupported scheme …") for those cases, so the gate eliminates the stuck-on-Checking failure mode without forcing me to catch `Error` subclasses (which `avoid_catching_errors` correctly flags). The pre-existing `on Exception` arm still covers timeouts, socket errors, etc.

File: lib/features/ai/ui/settings/services/connection_verifier_service.dart [hunk 2]
```diff
@@ -0,0 +1,362 @@
...
+  }) async {
+    final probes = ref.read(connectionProbeRegistryProvider);
+    final probe = probes[providerType];
+    if (probe == null) {
+      // Provider type with no registered probe (Whisper, Voxtral) —
+      // surface as idle rather than firing an HTTP call we can't form.
+      state = const ConnectionCheckIdle();
+      return;
+    }
+    if (apiKey.isEmpty && providerType != InferenceProviderType.ollama) {
```
[the remark this answers is not in the document]
Comment (cyberk1ng) [reply lib/features/ai/ui/settings/services/connection_verifier_service.dart]: Fixed. Switched to `apiKey.trim().isEmpty` so whitespace-only keys short-circuit to `Idle` instead of firing a probe that would just bounce on 401. The verifier now also defaults a blank `baseUrl` to `ProviderConfig.getDefaultBaseUrl(providerType)` upstream, matching the save path.

File: lib/features/ai/ui/settings/inference_provider_edit_page.dart [hunk 3]
```diff
@@ -588,11 +652,27 @@ class _InferenceProviderEditPageState
...
+                    providerType: formState.inferenceProviderType,
+                    apiKey: formController.apiKeyController.text,
+                    baseUrl: value,
+                  );
+                },
                 validator: (_) => formState.baseUrl.error?.displayMessage,
                 keyboardType: TextInputType.url,
               ),
             ),
+            SizedBox(height: tokens.spacing.step5),
```
[the remark this answers is not in the document]
Comment (cyberk1ng) [reply lib/features/ai/ui/settings/inference_provider_edit_page.dart]: Fixed. Wrapped the `SizedBox(height: tokens.spacing.step5)` in `if (ref.watch(connectionVerifierControllerProvider(formState.inferenceProviderType)) is! ConnectionCheckIdle)` so the gap only carves out when the strip is actually about to render content. When no probe has run the strip stays `SizedBox.shrink()` and the spacer is now skipped too — no more phantom whitespace between the base-URL field and the privacy hint.

File: lib/features/ai/ui/settings/inference_provider_edit_page.dart [hunk 4]
```diff
@@ -185,10 +195,58 @@ class _InferenceProviderEditPageState
...
+      const Duration(milliseconds: 600),
+      () {
+        if (!mounted) return;
+        ref
+            .read(
+              connectionVerifierControllerProvider(providerType).notifier,
+            )
+            .verify(baseUrl: baseUrl, apiKey: apiKey);
+      },
+    );
```
[the remark this answers is not in the document]
Comment (matthiasn) [reply lib/features/ai/ui/settings/inference_provider_edit_page.dart:239]: Fixed. Added `invalidate()` on `ConnectionVerifierController` that bumps the `_generation` guard without touching visible state, and the form's `_scheduleConnectionVerify` now calls it before scheduling the next debounce. Any in-flight probe still completes its HTTP roundtrip, but its post-await `state = result` write is dropped because `myGen != _generation` — no stale-credential flicker. Unit test added that holds a probe open via `Completer`, calls `invalidate()`, releases the probe, and asserts the state stays in `Checking` (i.e. the stale write did not land).

File: lib/features/ai/ui/settings/services/connection_verifier_service.dart [hunk 5]
```diff
@@ -0,0 +1,393 @@
...
+    // the `Checking` state forever. Reject upstream with an explicit
+    // network-failure message instead.
+    final scheme = parsedBase.scheme.toLowerCase();
+    if ((scheme != 'http' && scheme != 'https') || parsedBase.host.isEmpty) {
+      state = const ConnectionCheckFailedNetwork(
+        message:
+            'Base URL must include http(s) scheme and host '
+            '(e.g. https://api.example.com)',
+      );
+      return;
```
[the remark this answers is not in the document]
Comment (matthiasn) [reply lib/features/ai/ui/settings/services/connection_verifier_service.dart:238]: Fixed. Service layer no longer carries user-facing strings for the developer-written failures. Introduced a `ConnectionFailureCode` enum (`network` / `timeout` / `invalidBaseUrl` / `badResponseShape`) on `ConnectionCheckFailedNetwork`, added `aiProviderConnectionFailedInvalidBaseUrlDetail` to `app_en.arb` + all 5 other locale ARB files, and the UI now switches on `failed.code` in `_ConnectionStatusStrip` to pick the right localized detail. The dynamic platform exception messages (`SocketException` etc.) still flow through the generic `network` arm where raw is appropriate.

File: lib/features/ai/ui/settings/services/connection_verifier_service.dart [hunk 6]
```diff
@@ -0,0 +1,393 @@
...
+        // {"models": [...]} but some forks return a top-level list.
+        if (decoded is List) {
+          return ConnectionCheckVerified(
+            modelCount: decoded.length,
+            latency: stopwatch.elapsed,
+          );
+        }
+        return ConnectionCheckFailedNetwork(
+          message: 'Unexpected response shape: ${decoded.runtimeType}',
+        );
```
[the remark this answers is not in the document]
Comment (matthiasn) [reply lib/features/ai/ui/settings/services/connection_verifier_service.dart:326]: Fixed via the same `ConnectionFailureCode` dispatch — added `aiProviderConnectionFailedBadResponseDetail` (with `{type}` placeholder so the runtime type is interpolated) across all 6 locale ARB files. Service emits `ConnectionCheckFailedNetwork(message: "${decoded.runtimeType}", code: ConnectionFailureCode.badResponseShape)`; UI maps it via `messages.aiProviderConnectionFailedBadResponseDetail(failed.message)`.

File: lib/features/ai/ui/settings/services/connection_verifier_service.dart [hunk 7]
```diff
@@ -0,0 +1,393 @@
...
+      } on FormatException catch (e) {
+        return ConnectionCheckFailedNetwork(message: e.message);
+      }
+    }
+    return ConnectionCheckFailedHttp(
+      status: response.statusCode,
+      message: _extractErrorMessage(response.body),
+    );
+  } on TimeoutException {
+    return const ConnectionCheckFailedNetwork(message: 'Request timed out');
```
[the remark this answers is not in the document]
Comment (matthiasn) [reply lib/features/ai/ui/settings/services/connection_verifier_service.dart]: Fixed. `aiProviderConnectionFailedTimeoutDetail` added across all 6 locale ARB files. Service emits `ConnectionCheckFailedNetwork(message: '', code: ConnectionFailureCode.timeout)` on `TimeoutException`; UI dispatch in `_ConnectionStatusStrip` maps the code to the localized line. Unit test at line 277-280 now asserts on `code == ConnectionFailureCode.timeout` so the service contract stays honest.

File: lib/l10n/app_en.arb [hunk 8]
```diff
@@ -800,6 +800,48 @@
...
+      "count": {
+        "type": "int"
+      },
+      "ms": {
+        "type": "int"
+      }
+    }
+  },
+  "aiProviderConnectionVerifiedTitle": "Connection verified",
   "aiProviderConnectKeyHelperLink": "Get a key at {url}",
```
[the remark this answers is not in the document]
Comment (matthiasn) [reply lib/l10n/app_en.arb:855]: No-change on the original 8 keys — they're already present in `app_cs.arb`, `app_de.arb`, `app_es.arb`, `app_fr.arb`, and `app_ro.arb` (the original commit landed them everywhere; CodeRabbit's snapshot pre-dated those edits). The 3 new keys added by this round (`aiProviderConnectionFailedTimeoutDetail`, `aiProviderConnectionFailedInvalidBaseUrlDetail`, `aiProviderConnectionFailedBadResponseDetail`) are mirrored across all 6 locales with proper placeholders + informal-tone translations, and `make l10n` was re-run. Romanian uses the formal `dvs.` register; the new strings (`Cererea a expirat` / `URL-ul de bază trebuie să includă...` / `Formă de răspuns neașteptată`) match.

File: test/features/ai/ui/settings/inference_provider_edit_page_test.dart [hunk 9]
```diff
@@ -2710,4 +2715,305 @@ void main() {
...
+            additionalOverrides: verifierOverridesFor(probe: probe),
+          ),
+        );
+        await tester.pumpAndSettle();
+        await typeApiKeyAndFlushDebounce(tester);
+
+        final strings = l10n(tester);
+        expect(
+          find.text(strings.aiProviderConnectionFailedTitle('OpenAI')),
+          findsOneWidget,
```
[the remark this answers is not in the document]
Comment (matthiasn) [reply test/features/ai/ui/settings/inference_provider_edit_page_test.dart:2864]: Fixed at lines 2858-2862 and 2898-2902 of the test file — both `aiProviderConnectionFailedTitle('OpenAI')` assertions now read `aiProviderConnectionFailedTitle(strings.aiProviderOpenAiName)`, so the test stays correct if the OpenAI display copy ever changes. (The pre-existing tests at lines 357, 505, 519, etc. that match on `'OpenAI'` directly are outside this PR's scope; they'd be churn to touch and they're testing UI text that doesn't go through the title interpolation.)

File: lib/features/ai/ui/settings/services/connection_verifier_service.dart [hunk 10]
```diff
@@ -0,0 +1,438 @@
...
+    return const ConnectionCheckIdle();
+  }
+
+  /// Fires a probe against [baseUrl] using [apiKey]. Empty key or no
+  /// probe registered for [providerType] short-circuit to idle so
+  /// callers don't have to gate the call themselves.
+  Future<void> verify({
+    required String baseUrl,
+    required String apiKey,
+  }) async {
```
[the remark this answers is not in the document]
Comment (matthiasn) [reply lib/features/ai/ui/settings/services/connection_verifier_service.dart:179]: Fixed. `verify()` now trims the incoming key once into `normalizedKey` and uses it for both the empty-check guard AND the probe call. A pasted key with a trailing newline / stray space no longer ships the whitespace over the wire — auth uses the canonical credential.

File: lib/features/ai/ui/settings/services/connection_verifier_service.dart [hunk 11]
```diff
@@ -0,0 +1,438 @@
...
+    final trimmedBase = baseUrl.trim();
+    final effectiveBase = trimmedBase.isEmpty
+        ? ProviderConfig.getDefaultBaseUrl(providerType)
+        : trimmedBase;
+    final Uri parsedBase;
+    try {
+      parsedBase = Uri.parse(effectiveBase);
+    } on FormatException catch (e) {
+      state = ConnectionCheckFailedNetwork(message: e.message);
+      return;
```
[the remark this answers is not in the document]
Comment (matthiasn) [reply lib/features/ai/ui/settings/services/connection_verifier_service.dart:223]: Fixed. The `FormatException` arm on `Uri.parse(effectiveBase)` now emits `ConnectionCheckFailedNetwork(code: ConnectionFailureCode.invalidBaseUrl)` (same as the scheme/host guard below it) instead of leaking the raw Dart `FormatException.message` through the generic `network` arm. Users see the localized `Base URL must include http(s) scheme and host…` hint regardless of whether the URL fails to parse or just lacks structure.

File: test/features/ai/ui/settings/inference_provider_edit_page_test.dart [hunk 12]
```diff
@@ -2710,4 +2715,313 @@ void main() {
...
+
+        final strings = l10n(tester);
+        final retry = find.text(strings.aiProviderConnectionRetryButton);
+        await tester.ensureVisible(retry);
+        await tester.pumpAndSettle();
+        await tester.tap(retry);
+        await tester.pumpAndSettle();
+        // Second probe fired immediately, no debounce wait.
+        expect(probe.calls, 2);
+      },
```
[the remark this answers is not in the document]
Comment (matthiasn) [reply test/features/ai/ui/settings/inference_provider_edit_page_test.dart:3069]: Fixed. After `tester.tap(retry)` the test now does a single `tester.pump()` (no time advance) and asserts `probe.calls == 2` — that locks in the synchronous-fire behavior. Then advances 700 ms and re-asserts `probe.calls == 2` to catch any accidental debounce reintroduction (would surface as a second probe firing late).
